File: A19Sorting/A1BubbleSort.py


def bubble_sort(arr):
    for turn in range(len(arr)):
         for val in range(len(arr)-1-turn):
             if arr[val] > arr[val +1]:
                 temp = arr[val]
                 arr[val] = arr[val+1]
                 arr[val+1] = temp
    print(arr)
    
arr=[5,4,1,3,2,7]

bubble_sort(arr)

File: A19Sorting/A1BubbleSort.js
let arr = [5, 4, 1, 3, 2];

function bubbleSort(arr) {
  for (turn = 0; turn < arr.length; turn++) {
    for (let j = 0; j < arr.length - 1 - turn; j++) {
      if (arr[j] > arr[j + 1]) {
        temp = arr[j];
        arr[j] = arr[j + 1];
        arr[j + 1] = temp;
      }
    }
  }
  console.log(arr);
}

bubbleSort(arr);
